frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsAndSoil.tsx

```diff
@@ -109,6 +110,17 @@ const FieldsAndSoilComponent: FC<InputModuleProps> = ({
   const addFieldData = (values: FieldDetailInterface): void => {
     setTimeout(() => {
       const farmInfo: FarmDetailsInterface = { ...farmDetails };
+      const noSoilTestVal = {
+        ...values.SoilTest,
+        ValP: Infinity,
```

Comment:
just a random question, what does Infinity do?

Reply:
We use ranges for the calculation, which means that infinity here will always be greater than the min range it's being checkes for. It's used when the user does not provide soil/leaf tests